org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java

```diff
@@ -3974,6 +3974,11 @@ private void completionOnSingleNameReference(ASTNode astNode, ASTNode astNodePar
 					}
 					System.arraycopy(keywords, 0, keywords = new char[count][], 0, count);
 					findKeywords(this.completionToken, keywords, false, false);
+
+					// list all variables in the current scope which are visible here
+					checkCancel();
+					findVariablesAndMethods(this.completionToken, scope, singleNameReference, scope, insideTypeAnnotation,
+							singleNameReference.isInsideAnnotationAttribute, true, new ObjectVector());
```

Comment:
I believe this branch serves for proposals within a case label, right. For the current test we should never get here. Mind you, within a case label proposing variables and methods would be wrong.

We arrive here only because `astNodeParent` is a `SwitchStatement`, but since we are inside a `Block` the switch should not be our parent.

Reply:
I agree